mesa_frames/concrete/agentset.py

```
@@ -115,33 +117,46 @@
+                        f"Length of data ({len(agents)}) must match the number of columns in the AgentSet (excluding unique_id): {expected_columns}"
+                    )
+                new_agents = pl.DataFrame(
+                    [list(agents)],
+                    schema=[col for col in obj._agents.schema if col != "unique_id"],
+                    orient="row",
                 )
-            new_agents = pl.DataFrame([agents], schema=obj._agents.schema)
 
-        if new_agents["unique_id"].dtype != pl.Int64:
-            raise TypeError("unique_id column must be of type int64.")
+        new_agents = new_agents.with_columns(
```

Pylint (3.3.7)

[error] Using variable 'new_agents' before assignment

(E0601)